x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/SettingsPage/SettingsPage.tsx

```diff
@@ -0,0 +1,233 @@
+            if (didClickEnter && isFormValid) {
+              e.preventDefault();
+              handleSubmitEvent();
+            }
+          }}
+        >
+          {/* Selected Service panel */}
+          <EuiPanel paddingSize="m">
+            <EuiTitle size="s">
+              <h3>
+                {i18n.translate('xpack.apm.agentConfig.chooseServiceTitle', {
+                  defaultMessage: 'Choose service'
+                })}
+              </h3>
+            </EuiTitle>
+
+            <EuiSpacer size="m" />
+
+            <EuiFlexGroup>
+              <EuiFlexItem>
+                <EuiStat
+                  titleSize="xs"
+                  title={getOptionLabel(newConfig.service.name)}
+                  description="Service name"
+                />
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiStat
+                  titleSize="xs"
+                  title={getOptionLabel(newConfig.service.environment)}
+                  description="Environment"
+                />
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                {!isEditMode && (
+                  <EuiButton onClick={onClickEdit} iconType="pencil">
+                    {i18n.translate('xpack.apm.agentConfig.editButton', {
+                      defaultMessage: 'Edit'
+                    })}
+                  </EuiButton>
+                )}
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiPanel>
+
+          <EuiSpacer size="m" />
+
+          {/* Settings panel */}
+          <EuiPanel paddingSize="m">
+            <EuiTitle size="s">
+              <h3>
+                {i18n.translate('xpack.apm.agentConfig.settings.title', {
+                  defaultMessage: 'Core configuration options'
+                })}
+              </h3>
+            </EuiTitle>
+
+            <EuiSpacer size="m" />
+
+            {configSettingDefinitions
+
+              // filter out agent specific items that are not applicable
+              // to the selected service
+              .filter(filterByAgent(newConfig.agent_name as AgentName))
+              .map(setting => (
+                <SettingFormRow
+                  isUnsaved={unsavedChanges.hasOwnProperty(setting.key)}
+                  key={setting.key}
+                  setting={setting}
+                  value={newConfig.settings[setting.key]}
+                  onChange={(key, value) => {
+                    setNewConfig(prev => ({
+                      ...prev,
+                      settings: {
+                        ...prev.settings,
+                        [key]: value
+                      }
+                    }));
+                  }}
+                />
+              ))}
+          </EuiPanel>
+        </form>
+      </EuiForm>
+      <EuiSpacer size="xxl" />
+
+      {/* Bottom bar with save button */}
+      {unsavedChangesCount > 0 && (
+        <EuiBottomBar paddingSize="s">
```

Comment:
Shouldn't this be handled by EUI? @chandlerprall

Reply:
This can't be handled by EUI, because it's specific to application layouts. Not all consumers use the EuiNavDrawer or docking status. @timroes And I have briefly mentioned needing a global service for this like we do with toasts and modals. Perhaps now is a good time to actually investigate one?

Reply:
Maybe it could take a prop like `hasNavDrawer` or similar, which would add the styles @formgeist mentioned.